include/tlapack/lapack/getrf.hpp

```diff
@@ -0,0 +1,92 @@
+/// @file geqr2.hpp
+/// @author Ali Lotfi, University of Colorado Denver, USA
+/// Adapted from @see https://github.com/langou/latl/blob/master/include/geqr2.h
+//
+// Copyright (c) 2013-2022, University of Colorado Denver. All rights reserved.
+//
+// This file is part of <T>LAPACK.
+// <T>LAPACK is free software: you can redistribute it and/or modify it under
+// the terms of the BSD 3-Clause license. See the accompanying LICENSE file.
+
+#ifndef TLAPACK_GETRF_HH
+#define TLAPACK_GETRF_HH
+
+#include "tlapack/base/utils.hpp"
+
+namespace tlapack {
+
+/** Computes a QR factorization of a matrix A.
+ * 
+ * The matrix Q is represented as a product of elementary reflectors
+ * \[
+ *          Q = H_1 H_2 ... H_k,
+ * \]
+ * where k = min(m,n). Each H_i has the form
+ * \[
+ *          H_i = I - tau * v * v',
+ * \]
+ * where tau is a scalar, and v is a vector with
+ * \[
+ *          v[0] = v[1] = ... = v[i-1] = 0; v[i] = 1,
+ * \]
+ * with v[i+1] through v[m-1] stored on exit below the diagonal
+ * in the ith column of A, and tau in tau[i].
+ * 
+ * @return  0 if success
+ * 
+ * @param[in,out] A m-by-n matrix.
+ *      On exit, the elements on and above the diagonal of the array
+ *      contain the min(m,n)-by-n upper trapezoidal matrix R
+ *      (R is upper triangular if m >= n); the elements below the diagonal,
+ *      with the array tau, represent the unitary matrix Q as a
+ *      product of elementary reflectors.
+ * @param[out] tau Real vector of length min(m,n).
+ *      The scalar factors of the elementary reflectors.
+ *      If
+ *          n-1 < m and
+ *          type_t<matrix_t> == type_t<vector_t>
+ *      then one may use tau[1:n] as the workspace.
+ * 
+ * @param work Vector of size n-1.
+ * 
+ * @ingroup geqrf
```

Comment:
change. + speak to Weslley, I do not think we should d ``group`` anyway.

Reply:
remove this ``@ingroup geqrf`` line

Reply:
Hi Julien. Currently, we are using `@ingroup` to organize the routines in the documentation. Look the way they are split now: https://tlapack.github.io/. This is the same in LAPACK++: https://icl.bitbucket.io/lapackpp/.

(Yes, I put the documentation online! It was trivial to do.)